cli/pd_cli.py

```diff
@@ -439,6 +462,171 @@ def usage(self, exception, command):
     print >> sys.stderr, exception
     print >> sys.stderr, "Try 'help %s' for more information" % command
 
+#-------------- LAKI -------------
+# Multicast API
+  def do_mc_mgrp_create(self, line):
+    """
+    mc_mgrp_create MGRP_ID
+    This function creates a multicast group with multicast index MGRP_ID.
+    For example: mc_mgrp_create 1
+    """
+    words = collections.deque(line.split())
+    try:
+       mgid = self.get_handle(words, "MGRP_ID")
+       mgrp_hdl = self._thrift_client.mc_mgrp_create(mgid)
+       print "MGRP has created with handle %d" % mgrp_hdl
+    except NameError as ne:
+      print >> sys.stderr, ne
+    except thrift.protocol.TProtocol.TProtocolException as e:
+      print >> sys.stderr, e
+    except Exception as e:
+      self.usage(e, "mc_mgrp_create")
+
+
+  def do_mc_l1_node_create(self, line):
+    """
+    mc_l1_node_create R_ID
+    This function creates a L1 multicast node with replication id R_ID
+    For example: mc_l1_node_create 0
+    """
+    words = collections.deque(line.split())
+    try:
+       rid = self.get_handle(words, "R_ID")
+       l1_hdl = self._thrift_client.mc_l1_node_create(rid)
+       print "L1 has created with handle %d" % l1_hdl
+    except NameError as ne:
+      print >> sys.stderr, ne
+    except thrift.protocol.TProtocol.TProtocolException as e:
+      print >> sys.stderr, e
+    except Exception as e:
+      self.usage(e, "mc_l1_node_create")
+
+  def do_mc_l1_associate_node(self, line):
+    """
+    mc_l1_associate_node MGRP_HANDLE L1_HANDLE
+    This function associates a L1 node to a multicast tree.
+    For example: mc_l1_associate_node 1234321 1234567
+    """
+    words = collections.deque(line.split())
+    try:
+       mgrp_hdl = self.get_handle(words, "MGRP_HANDLE")
+       l1_hdl = self.get_handle(words, "L1_HANDLE")
+       self._thrift_client.mc_l1_associate_node(mgrp_hdl, l1_hdl)
+       print "L1 has been assiciated to the multicast group"
+    except NameError as ne:
+      print >> sys.stderr, ne
+    except thrift.protocol.TProtocol.TProtocolException as e:
+      print >> sys.stderr, e
+    except Exception as e:
+      self.usage(e, "mc_l1_associate_node")
+  
+  ## TODO - add multiple port and lag support
+  def do_mc_l2_node_create(self, line): #l1_hdl, port_map, lag_map):
+    """
+    mc_l2_node_create L1_HANDLE PORT_MAP LAG_MAP
+    This function creates a Level 2 multicast node with list of ports defined by PORT_MAP (as a bit vector)
+    For exmaple: mc_l2_node_create 1234567 30 -1
+    If PORT_MAP or LAG_MAP is -1, their value is not specified.
+    """
+    words = collections.deque(line.split())
+    try:
+       l1_hdl = self.get_handle(words, "L1_HANDLE")
+       port_map = self.get_handle(words, "PORT_MAP")
+       lag_map = self.get_handle(words, "LAG_MAP")
+       ports = [0] * 32
+       lags = [0] * 32
+       if port_map != -1:
+         ports[0] = port_map
```

Comment:
do this mean that we can only configure ports 0 through 7 (given that ports is a byte array)
this is acceptable for now, but it would be good if it were documented with a comment in the code, and if there was a check to verify that the user specified value is less than 256